modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -68,6 +72,7 @@
             }
             String[] types = extractStringArray(source, "type");
             if (types != null) {
+                deprecationLogger.deprecatedAndMaybeLog("reindex_with_types", TYPES_DEPRECATION_MESSAGE);
```

Comment:
For me it'd be clearer to move these checks out of the parsing logic into `buildRequest`, where you can examine the parsed `ReindexRequest`. This is just a personal preference though, totally up to you if you'd like to change.

Reply:
I took a look at the alternative but I think I prefer it here